specifyweb/frontend/js_src/lib/components/FormEditor/Editor.tsx

```diff
@@ -113,7 +113,7 @@ export function FormEditorWrapper(): JSX.Element {
               const newViewDefs = viewSets.viewDefs.filter(
                 (viewDefinition) =>
                   currentUsedViewDefinitions.has(viewDefinition.name) &&
-                  !updatedUsedViewDefinitions.has(viewDefinition.name)
+                  updatedUsedViewDefinitions.has(viewDefinition.name)
               );
 
               setViewSets(
```

Comment:
seems like I got confused here - the comment above is correct, but I forgot that .filter() in this case is not deleting view definitions, but deciding which definitions should remain (i.e not get deleted), thus the condition should have been reversed:

> delete all views that were not used before and are used now

turns into

> preserve all views that either weren't used before, or are used now

your code change would make it delete all view definitions that aren't used now OR weren't used before - I would say that's potentially unsafe and maybe unexpected to the user (why deleting this view def caused other view defs to disappear). this is a reasonable behavior for a new system, but here we are adding a UI to a very old system, will all kinds of messy data

for all we know, some views are marked as unused only because the user made a typo in the name somewhere
OR maybe instead of commenting out a view, the user just renamed it to an unused name

whatever it is, we probably don't want to delete all unused, which is what the code after your code change does